common/spec/dependabot/file_fetchers/vendor_updater_spec.rb

```diff
@@ -0,0 +1,88 @@
+# frozen_string_literal: true
+
+require "spec_helper"
+require "dependabot/file_updaters/vendor_updater"
+
+RSpec.describe Dependabot::FileUpdaters::VendorUpdater do
+  let(:updater) do
+    described_class.new(
+      repo_contents_path: repo_contents_path,
+      vendor_dir: vendor_dir
+    )
+  end
+
+  let(:vendor_dir) { File.join(repo_contents_path, "vendor/cache") }
+  let(:project_name) { "vendor_gems" }
+  let(:repo_contents_path) { build_tmp_repo(project_name) }
+
+  let(:updated_files) do
+    updater.updated_vendor_cache_files(base_directory: repo_contents_path)
+  end
+
+  before do
+    in_cloned_repository(repo_contents_path) do
+      `mv vendor/cache/business-1.4.0.gem vendor/cache/business-1.5.0.gem`
```

Comment:
Bit of cheating here, but I didn't want these tests to depend on any specific updater implementation, so just testing that updated files in the right folder are picked up

Reply:
```suggestion
      # change a vendor file like an updater would
      `mv vendor/cache/business-1.4.0.gem vendor/cache/business-1.5.0.gem`
```

Idea: encode that magic.